server/container/util-java8/src/main/java/org/apache/james/util/FluentFutureStream.java

```diff
@@ -69,11 +80,28 @@ private FluentFutureStream(CompletableFuture<Stream<T>> completableFuture) {
             CompletableFutureUtil.thenComposeOnAll(completableFuture(), function));
     }
 
+    public <U> FluentFutureStream<U> thenFlatCompose(Function<T, CompletableFuture<Stream<U>>> function) {
```

Comment:
Ok, now we need documentation because thenCompose is supposed to be a kind of flatMap already, what does theFlatCompose do ?
There's several solutions : 
1. every methods here are legitimate and then we have to write some javadoc because I can't understand them
2. some methods don't make sense  anymore and we can tidy things for normal people to understand
3. we rething the API to stick to map and flatmap so that everything becomes easier for people already understanding map/flatmap (in Scala, Future is a monad and I don't have to understand new concept to use them)